doc/source/advanced-use.rst

```diff
@@ -11,251 +16,317 @@ Essentially, clones are cheap, lightweight, and writable snapshots.
 A clone depends on its source snapshot and filesystem. To destroy the
 source jail and preserve its clones, the clone must be promoted first.
 
-Create a clone
+.. index:: Create clones
+.. _Create a Clone:
+
+Create a Clone
 ++++++++++++++
 
 To clone **www01** to **www02**, run:
 
-:samp:`# iocage clone www01 tag=www02`
+:samp:`# iocage clone www01 --name www02`
 
 Clone a jail from an existing snapshot with:
 
-:samp:`# iocage clone www01@snapshotname tag=www03`
+:samp:`# iocage clone www01@snapshotname --name www03`
+
+.. index:: Promote a Clone
+.. _Promoting a Clone:
 
-Promoting a clone
+Promoting a Clone
 +++++++++++++++++
 
+.. warning:: This functionality isn't fully added to iocage, and may not
+   function as expected.
+
 **To promote a cloned jail, run:**
 
-:samp:`# iocage promote UUID | TAG`
+:command:`iocage promote [UUID | NAME]`
 
 This reverses the *clone* and *source* jail relationship. The clone
 becomes the source and the source jail is demoted to a clone.
 
 **The demoted jail can now be removed:**
 
-:samp:`# iocage destroy UUID | TAG`
+:command:`iocage destroy [UUID | NAME]`
+
+.. index:: Updating Jails
+.. _Updating Jails:
 
-Updating jails
+Updating Jails
 --------------
 
 Updates are handled with the freebsd-update(8) utility. Jails can be
-updated while they are stopped or running.
+updated while they are stopped or running. While updating can seem
+routine, it is always recommended to use ZFS snapshot functionality to
+create a backup of the jail before updating.
 
-To update a jail to latest patch level, run:
+Create a backup snapshot of the jail:
 
-:samp:`# iocage update UUID | TAG`
+:command:`iocage snapshot -n [snapshotname] [UUID | NAME]`
 
-This automatically creates a back-out snapshot of the jail.
+To update a jail to latest patch level, run:
+
+:command:`iocage update [UUID | NAME]`
 
 When updates are finished and the jail appears to function properly,
 remove the snapshot:
 
-:samp:`# iocage snapremove UUID|TAG@snapshotname`
-
-If the update breaks the jail, revert back to the original snapshot:
-
-:samp:`# iocage rollback UUID|TAG@snapshotname`
+:command:`iocage snapremove [UUID|NAME]@[snapshotname]`
 
 To test updating without affecting a jail, create a clone and update the
 clone the same way as outlined above.
 
 To clone a jail, run:
 
-:samp:`# iocage clone UUID|TAG tag=testupdate`
+:command:`iocage clone [UUID|NAME] --name [testupdate]`
```

Comment:
We should note that `--name` is optional in this case. If they don't supply it, they will just get a UUID named jail.